src/Server/HTTP/HTTPServerResponse.cpp

```diff
@@ -166,8 +125,31 @@ void HTTPServerResponse::redirect(const std::string & uri, HTTPStatus status)
 
     // Send header
     Poco::Net::HTTPHeaderOutputStream hs(session);
-    write(hs);
+    beginWrite(hs);
+    hs << "\r\n";
     hs.flush();
 }
 
+void HTTPServerResponse::allowKeepAliveIFFRequestIsFullyRead()
+{
+    /// HTTPChunkedReadBuffer doesn't consume the final \r\n0\r\n\r\n if the caller reads exactly all bytes,
+    /// without checking for eof() (i.e. trying to read past the end) after that.
+    /// If those leftovers are then seen by the next request's HTTPServerRequest::readRequest,
+    /// it would in theory produce the error:
```

Comment:
Not just in theory anymore, `test_http_connection_drain_before_reuse` hits this.
```suggestion
    /// it would produce the error:
```